prow/cmd/hook/main.go

```diff
@@ -136,7 +136,6 @@ func main() {
 	if err != nil {
 		logrus.WithError(err).Fatal("Error getting Git client.")
 	}
-	defer gitClient.Clean()
```

Comment:
Why do we even do the `gitClient.Clean()`? The only case in which this will make any difference is if the app crashes unexpectedly, because in all other cases the container gets removed, including the git data. And if the app crashed unexpectedly, we probably have other problems anyways.

Reply:
I guess it's nice for local runs?